File: HealthAxis.Api/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;

namespace HealthAxis.Api.Models
{
    public class Doctor
    {
        [Key]
        public int DoctorId { get; set; }
        [Required]
        [MinLength(2)]
        [RegularExpression(@"^[A-Z][A-Za-z\s]+$")]
        public required string FullName { get; set; }
        [Required]
        [RegularExpression("(Endocrinologist|Oncologist|Gynecologist|OrthopedicSurgeon|Psychiatrist|Pediatrician|Neurologist|Dermatologist|Cardiologist|GeneralPractitioner)", ErrorMessage = "Invalid Specialisation")]
        public required string Specialisation { get; set; }
        [Required]
        [Range(0, 70, ErrorMessage = "Years of experience should be between o to 70")]

        public int YearsOfExperience { get; set; }
        [Required]
        [Range(0, 100000)]
        public int ConsultationFee { get; set; }
        [Required]
        public bool IsActive { get; set; }
        public string UserId { get; set; } = string.Empty;
    }
}

File: HealthAxis.Angular/src/app/dtos/doctor.ts
import { Specialisation } from '../enums/specialisation';

export interface Doctor {

  doctorId: number;

  fullName: string;

  email: string;

  specialisation: Specialisation;

  yearsOfExperience: number;

  consultationFee: number;

  isActive: boolean;
}

export interface UpdateDoctorRequest {

  doctorId: number;

  fullName: string;

  specialisation: Specialisation;

  yearsOfExperience: number;

  consultationFee: number;

  isActive: boolean;
}